### PR TITLE
Switched to codecov from codeclimate for test coverage

### PR DESCRIPTION
**Pull request by gsaslis**
**What this PR does / why we need it**:

Switch from codeclimate.com to codecov.com for test coverage, as per other 3scale projects. 

**Which issue(s) this PR fixes** 

test coverage broken on code climate

**Verification steps** 

- [x] circleci should run on this PR
- [x] the code coverage report should appear on codecov.com
- [x] once integration with the GH app is enabled, a comment should appear on this PR

**Special notes for your reviewer**:

Uses both ruby gem + codecov bash uploader, since some parts of the test suite are running without internet access (on purpose) and we cannot upload test coverage reports to codecov through its ruby gem, even though, by default, it handles that out of the box.
Instead, we add an extra step to upload after internet access has been restored.

### CODE REVIEW

**Comment by gsaslis** [comment]
@mikz can you please enable codecov GitHub app on this repo?

**Comment by mikz** [comment]
@gsaslis done

**Comment by gsaslis** [comment]
@mikz hmm circleci checks fail because we have disabled passing secrets to builds on PRs from forks. 

which means that no external contributor builds can work here without either: 
* enabling that (a security risk)
* proceeding to merge #44 

/cc @3scale/system

**Comment by mikz** [comment]
@gsaslis Yep. That is expected. #44 is the way to go.

**Comment by gsaslis** [comment]
ok, so:

blocked on #44

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -1,5 +1,12 @@
-require 'codeclimate_rails'
-CodeclimateRails.start
+
+require 'simplecov'
+SimpleCov.start
```
Comment (mikz) [inline spec/spec_helper.rb]: This should happen only when `CI` is set. No need to measure coverage locally.

**Comment by mikz** [comment]
You might want some setup to make codecov less annoying: https://github.com/3scale/apicast/blob/master/.codecov.yml

File: test/test_helper.rb [hunk 2]
```diff
@@ -4,9 +4,13 @@
 
 require File.expand_path(File.dirname(__FILE__) + "/../config/environment")
 
-require 'codeclimate_rails'
-CodeclimateRails.start
+if ENV['CI']
+  require 'simplecov'
+  SimpleCov.start
```
[the remark this answers is not in the document]
Comment (gsaslis) [reply test/test_helper.rb]: @mikz if I understand correctly that means above the previous line `require File.expand_path(File.dirname(__FILE__) + "/../config/environment")` ?

[the remark this answers is not in the document]
Comment (mikz) [reply test/test_helper.rb]: Yes. To be the first thing loaded.

**Comment by gsaslis** [comment]
btw, this passed all tests on circle https://circleci.com/workflow-run/818bfd04-23d7-411b-89e4-bc984c4a7aa9

(it just didn't show here as I'm doing some other work on my fork and you can't have builds on both upstream and your fork, so... )

@mikz I think we're good to merge this one...?

**Pull request merged**